diff --git a/orbit/cmd/orbit/orbit.go b/orbit/cmd/orbit/orbit.go
--- a/orbit/cmd/orbit/orbit.go
+++ b/orbit/cmd/orbit/orbit.go
@@ -92,6 +92,11 @@ func main() {
 		shellCommand,
 	}
 	app.Flags = []cli.Flag{
+		&cli.StringFlag{
+			Name:    "osqueryd-path",
+			Usage:   "Override the path to the osqueryd binary",
+			EnvVars: []string{"NIX_ORBIT_OSQUERYD_PATH"},
+		},
 		&cli.StringFlag{
 			Name:    "root-dir",
 			Usage:   "Root directory for Orbit state",
@@ -603,7 +608,11 @@ func main() {
 		// it fetches osqueryd once as part of initialization.
 		var updater *update.Updater
 		var updateRunner *update.Runner
-		if !c.Bool("disable-updates") || c.Bool("dev-mode") {
+
+		if override := c.String("osqueryd-path"); override != "" {
+			log.Info().Msgf("Overriding osqueryd path: %s", override)
+			osquerydPath = override
+		} else if !c.Bool("disable-updates") || c.Bool("dev-mode") {
 			updater, err := update.NewUpdater(opt)
 			if err != nil {
 				return fmt.Errorf("create updater: %w", err)
